google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcConversions.java

```diff
@@ -714,8 +714,12 @@ private BucketInfo.LifecycleRule lifecycleRuleDecode(Bucket.Lifecycle.Rule from)
               .collect(ImmutableList.toImmutableList());
       conditionBuilder.setMatchesStorageClass(collect);
     }
-    conditionBuilder.setMatchesPrefix(condition.getMatchesPrefixList());
-    conditionBuilder.setMatchesSuffix(condition.getMatchesSuffixList());
+    if (!condition.getMatchesPrefixList().isEmpty()) {
```

Comment:
this change looks out of context, is it needed as well?

Reply:
It's a difference between what is constructed via 
```
 LifecycleRule d1 =
        new LifecycleRule(
            LifecycleAction.newAbortIncompleteMPUploadAction(),
            LifecycleCondition.newBuilder().setAge(0).build());
```

And what is decoded coming back from the service via grpc. By simply setting as we were doing before the value would be empty list rather than null. Apiary on the other hand will be null by default. This change brings these both in line with the apiary behavior.